### PR TITLE
GitHub proxy part 4: `tsh git ls` with unified resource

### PR DESCRIPTION
**Pull request by greedy52**
related:
- #48762

sample output:
```
$ tsh git ls
Type   Organization         Username URL                                     
------ -------------------- -------- --------------------------------------- 
GitHub goteleport-core-test (n/a)*   https://github.com/goteleport-core-test 

(n/a)*: Username will be retrieved automatically upon running git commands.
        Alternatively, run `tsh git login --github-org <org>`.

hint: use 'tsh git clone <git-clone-ssh-url>' to clone a new repository
      use 'tsh git config update' to configure an existing repository to use Teleport
      once the repository is cloned or configured, use 'git' as normal
```

### CODE REVIEW

File: api/types/resource.go [hunk 1]
```diff
@@ -139,6 +139,18 @@ type EnrichedResource struct {
 	RequiresRequest bool
 }
 
+// EnrichedResources is a list of Enriched resources.
```
Comment (zmb3) [inline api/types/resource.go:142]: But what is an enriched resource?

Comment (greedy52) [reply api/types/resource.go]: i have changed it to `EnrichedResources is a wrapper of []*EnrichedResource.`. any other recommendations?

File: tool/tsh/common/git_list.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	if showLoginNote {
+		fmt.Fprint(cf.Stdout(), gitLoginNote)
+	}
+
+	fmt.Fprint(cf.Stdout(), gitCommandsGeneralHint)
+	return nil
+}
+
+const gitLoginNote = "" +
+	"(n/a)*: Usernames will be retrieved automatically upon git commands.\n" +
```
Comment (r0mant) [inline tool/tsh/common/git_list.go:168]:
```suggestion
	"(n/a)*: Username will be retrieved automatically upon running git commands.\n" +
```

**Review by r0mant — APPROVED** [review]

File: api/types/resource.go (hunk 1 above)
Comment (avatus) [reply api/types/resource.go]: [EnrichedResource](https://github.com/gravitational/teleport/blob/master/api/types/resource.go#L131-L140) is a resource that has had `Logins` and `RequiresRequest` added to it.

**Review by avatus — APPROVED** [review]

**Pull request merged**